nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -761,24 +771,38 @@ private Request configureRequest(final ProcessContext context, final ProcessSess
     }
 
     private RequestBody getRequestBodyToSend(final ProcessSession session, final ProcessContext context, final FlowFile requestFlowFile) {
-        return new RequestBody() {
-            @Override
-            public MediaType contentType() {
-                String contentType = context.getProperty(PROP_CONTENT_TYPE).evaluateAttributeExpressions(requestFlowFile).getValue();
-                contentType = StringUtils.isBlank(contentType) ? DEFAULT_CONTENT_TYPE : contentType;
-                return MediaType.parse(contentType);
-            }
+        if(context.getProperty(PROP_SEND_BODY).asBoolean()) {
+            return new RequestBody() {
+                @Override
+                public MediaType contentType() {
+                    String contentType = context.getProperty(PROP_CONTENT_TYPE).evaluateAttributeExpressions(requestFlowFile).getValue();
+                    contentType = StringUtils.isBlank(contentType) ? DEFAULT_CONTENT_TYPE : contentType;
+                    return MediaType.parse(contentType);
+                }
 
-            @Override
-            public void writeTo(BufferedSink sink) throws IOException {
-                session.exportTo(requestFlowFile, sink.outputStream());
-            }
+                @Override
+                public void writeTo(BufferedSink sink) throws IOException {
+                    session.exportTo(requestFlowFile, sink.outputStream());
+                }
 
-            @Override
-            public long contentLength(){
-                return useChunked ? -1 : requestFlowFile.getSize();
-            }
-        };
+                @Override
+                public long contentLength(){
+                    return useChunked ? -1 : requestFlowFile.getSize();
+                }
+            };
+        } else {
+            return new RequestBody() {
+                @Override
+                public void writeTo(BufferedSink sink) throws IOException {
+                    sink.writeUtf8("");
+                }
+
+                @Override
+                public MediaType contentType() {
+                    return MediaType.parse("");
+                }
+            };
```

Comment:
For this 'else' clause, it would probably be better to use a RequstBody static factory.  Something like:
`return RequestBody.create(null, new byte[0]);`

I'm not sure the proper way to specific an empty MediaType, though. I think reading the MediaType javadoc, my hunch is that MediaType.parse("") should return null (since empty string is an invalid type).